### PR TITLE
[Tests-Only] ideas for a tool to make BDD for OCIS using oc10 API tests easier

### PR DESCRIPTION
**Pull request by individual-it**
ideas for a tool that would make the multi-stage BDD process described in https://owncloud.github.io/extensions/ocis_reva/testing/ simpler.

@jnweiger @C0rby @kulmann @phil-davis @dpakach @jnweiger @LukasHirt @refs @PVince81 would a tool described in the feature files be helpful? Or what would be helpful to make your live easier with the API tests and the juggling of the multiple repos?

### CODE REVIEW

File: tests/acceptance/tools/BDDForOCIS/createPRs.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  I want to be able to create PRs to ocis & ocis-reva that run tests from the current core branch
+  So that I can confirm that the current tests pass in ocis & ocis-reva
+
+  As a ocis developer
+  I want to use the current oC10 API tests for BDD
+  So that I can make sure ocis is API compatible with oC10
+
+  Scenario: create new PRs
+    Given the user has checked out the branch "changed-tests"
+    When the user runs the command "crate-prs"
```
Comment (phil-davis) [inline tests/acceptance/tools/BDDForOCIS/createPRs.feature:13]:
```suggestion
    When the user runs the command "create-prs"
```

File: tests/acceptance/tools/BDDForOCIS/createPRs.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    Then a new PR to the "ocis" repo should be created with the core branch "changed-tests" as test-runner
+    And the link to the new PR in the "ocis" repo should be listed on the console
+    And a new PR to the "ocis-reva" repo should be created with the core branch "changed-tests" as test-runner
+    And the link to the new PR in the "ocis-reva" repo should be listed on the console
+
+  Scenario: update existing PRs
+    Given the user has checked out the branch "changed-tests"
+    And an open PR to the "ocis" repo exists with the core branch "changed-tests" as test-runner
+    And an open PR to the "ocis-reva" repo exists with the core branch "changed-tests" as test-runner
+    When the user runs the command "crate-prs"
```
Comment (phil-davis) [inline tests/acceptance/tools/BDDForOCIS/createPRs.feature:23]:
```suggestion
    When the user runs the command "create-prs"
```

File: tests/acceptance/tools/BDDForOCIS/createPRs.feature [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    Then the existing PR to the "ocis" repo should be rebased and CI should be retriggered
+    And the link to the existing PR in the "ocis" repo should be listed on the console
+    And the existing PR to the "ocis-reva" repo should be rebased and CI should be retriggered
+    And the link to the existing PR in the "ocis-reva" repo should be listed on the console
+
+  Scenario Outline: create new PRs with bugfix changes
+    Given the user has checked out the branch "changed-tests"
+    And the user has checked out the branch "bug-fix" of the "<repo-with-changes>" local clone
+    When the user runs the command "crate-prs --<repo-with-changes>=%path-to-<repo-with-changes>-clone%"
```
Comment (phil-davis) [inline tests/acceptance/tools/BDDForOCIS/createPRs.feature:32]:
```suggestion
    When the user runs the command "create-prs --<repo-with-changes>=%path-to-<repo-with-changes>-clone%"
```

**Comment by PVince81** [comment]
somehow with all the recent happenings I wonder if we should use git submodules or use the commit hash in the git clone command in Drone to run tests on a specific version of the tests

this would provide reproducible test runs, because now often times when tests were passing in one build it fails in another because owncloud/core was changed: https://github.com/owncloud/ocis-reva/pull/215#issuecomment-632586426

**Comment by mmattel** [comment]
Pls rename the PR and add a leading `[Tests-Only] ` as it targets tests.
For reference, see all the other test PRs.

**Comment by refs** [comment]
> Pls rename the PR and add a leading `[Tests-Only] ` as it targets tests.
> For reference, see all the other test PRs.

Thanks @mmattel !

**Comment by individual-it** [comment]
this is not needed anymore see discussion in https://jira.owncloud.com/browse/OCIS-217
an other way to simplify test runs was implemented see https://github.com/owncloud/core/pull/37717

**Pull request closed without merging**